### PR TITLE
Feature/friend invite

### PR DESCRIPTION
**Pull request by deeksha-developer**
https://app.clickup.com/t/85ztzbbvq-clickup link
code optimization
image

### CODE REVIEW

File: src/app/userDashboard/tournament/page.tsx [hunk 1]
```diff
@@ -19,16 +19,18 @@ import CountdownComponent from './CountdownComponent';
...
   gameType: string;
   mapType: string;
   version: string;
-  date: string;
-  time: string;
-  lastServival: string;
-  roomUuid: string;
+  dateAndTime: Date | string;
+  roomId: string;
+ // time: string;
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove comment

File: src/app/userDashboard/tournament/page.tsx [hunk 2]
```diff
@@ -19,16 +19,18 @@ import CountdownComponent from './CountdownComponent';
...
+ lastServival: string;
+ // roomUuid: string;
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove comment

File: src/app/userDashboard/tournament/page.tsx [hunk 3]
```diff
@@ -19,16 +19,18 @@ import CountdownComponent from './CountdownComponent';
...
   mapType: string;
   version: string;
-  date: string;
-  time: string;
-  lastServival: string;
-  roomUuid: string;
+  dateAndTime: Date | string;
+  roomId: string;
+ // time: string;
+ lastServival: string;
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: Fix spelling

File: src/app/userDashboard/tournament/page.tsx [hunk 4]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
-  const [gameName, setMatchName] = useState<String>('');
-  const [gameType, setGameType] = useState<String>('');
-  const [mapType, setMapType] = useState<String>('');
-  const [version, setVersion] = useState<String>('');
-  const [date, setDate] = useState<String>('');
-  const [time, setTime] = useState<String>('');
-  const [lastServival, setLastServival] = useState<String>('');
-  const [roomId, setRoomId] = useState<String>('');
-  const [mapImg, setMapImg] = useState<String>('');
+  const [gameName, setMatchName] = useState<string>('');
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this state we donot need

File: src/app/userDashboard/tournament/page.tsx [hunk 5]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
+  const [gameType, setGameType] = useState<string>('');
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this state we donot need

File: src/app/userDashboard/tournament/page.tsx [hunk 6]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
+  const [mapType, setMapType] = useState<string>('');
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this state we donot need

File: src/app/userDashboard/tournament/page.tsx [hunk 7]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
+  const [version, setVersion] = useState<string>('');
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this state we donot need

File: src/app/userDashboard/tournament/page.tsx [hunk 8]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
+  const [date, setDate] = useState<string>('');
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this state we donot need

File: src/app/userDashboard/tournament/page.tsx [hunk 9]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
+  const [time, setTime] = useState<string>("");
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this state we donot need

File: src/app/userDashboard/tournament/page.tsx [hunk 10]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
+  const [lastServival, setLastServival] = useState<string>('');
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this state we donot need

File: src/app/userDashboard/tournament/page.tsx [hunk 11]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
+  const [roomId, setRoomId] = useState<string>('');
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this state we donot need

File: src/app/userDashboard/tournament/page.tsx [hunk 12]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
+  const [mapImg, setMapImg] = useState<string>('');
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this state we donot need

File: src/app/userDashboard/tournament/page.tsx [hunk 13]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
   const [matchIndex, setMatchIndex] = useState<number[]>([]);
   const [isLoading, setIsLoading] = useState<boolean>(false);
   const [userName, setUserName] = useState<string>("");
+    const initialValues:tournament = {
+    gameName:"", mapType:"", gameType:"", version:"", dateAndTime:new Date(), lastServival:"" , roomId:"", mapImg:""
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: lastServival spelling mistake survival

File: src/app/userDashboard/tournament/page.tsx [hunk 14]
```diff
@@ -38,19 +40,25 @@ function Tournament() {
...
+  }
+  const [matchDetails , setMatchDetails] = useState<tournament>(initialValues)
+  
 
+console.log("matchDetails",matchDetails)
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove this console

File: src/app/userDashboard/tournament/page.tsx [hunk 15]
```diff
@@ -111,22 +120,40 @@ function Tournament() {
 
   useEffect(() => {
     setLastTournament(alldata[alldata.length - 1]);
-    setAllTournaments(alldata?.slice(0, 8));
+    setAllTournaments(alldata?.slice(0, 50));
     getRegisteredMatches();
     getRoomidPwd();
   }, [alldata]);
 
+  // useEffect(() => {
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove comment

File: src/app/userDashboard/tournament/page.tsx [hunk 16]
```diff
@@ -135,23 +162,22 @@ function Tournament() {
     gType: string,
     mType: string,
     vType: string,
-    mdate: string,
+    mdate:string,
+    dateandtime: Date,
     mtime: string,
     lastServival: string,
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: spelling mistake

File: src/app/userDashboard/tournament/page.tsx [hunk 17]
```diff
@@ -135,23 +162,22 @@ function Tournament() {
     gType: string,
     mType: string,
     vType: string,
-    mdate: string,
+    mdate:string,
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: what is this mDate we are not using this now

File: src/app/userDashboard/tournament/page.tsx [hunk 18]
```diff
@@ -135,23 +162,22 @@ function Tournament() {
...
+    dateandtime: Date,
     mtime: string,
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: time we are not using

File: src/app/userDashboard/tournament/page.tsx [hunk 19]
```diff
@@ -195,10 +221,8 @@ function Tournament() {
       console.log('Failed to sign up. Please try again.');
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove console

File: src/app/userDashboard/tournament/page.tsx [hunk 20]
```diff
@@ -248,21 +272,28 @@ function Tournament() {
...
     const [hour, minutes] = (time || "").split(":").map((values) => parseInt(values));
-    const [year, month, matchData] = (date || "").split("-").map((values) => parseInt(values));
-    let dateNumber = new Date(year, month - 1, matchData, hour, minutes).getTime();
+  const [year, month, matchData] = (date || "").split("-").map((values) => parseInt(values));
+   let dateNumber = new Date(year, month - 1, matchData, hour, minutes).getTime();
     const reducedTime = new Date(dateNumber - REDUCE_TIME).getTime();
     if (currentTime >= reducedTime) {
-      return { roomId, password }
+    
+       return { roomId:"i2334", password:'22344'}
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: this has to be dynamic

File: src/app/userDashboard/tournament/page.tsx [hunk 21]
```diff
@@ -38,18 +39,23 @@ function Tournament() {
...
-  const [gameName, setMatchName] = useState<String>('');
-  const [gameType, setGameType] = useState<String>('');
-  const [mapType, setMapType] = useState<String>('');
-  const [version, setVersion] = useState<String>('');
-  const [date, setDate] = useState<String>('');
-  const [time, setTime] = useState<String>('');
-  const [lastServival, setLastServival] = useState<String>('');
-  const [roomId, setRoomId] = useState<String>('');
-  const [mapImg, setMapImg] = useState<String>('');
+  // const [gameName, setMatchName] = useState<string>('');
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: Remove comments

File: src/app/userDashboard/tournament/page.tsx [hunk 22]
```diff
@@ -38,18 +39,23 @@ function Tournament() {
...
+  // const [date, setDate] = useState<string>('');
+  // const [time, setTime] = useState<string>("");
+  // const [lastServival, setLastServival] = useState<string>('');
+  // const [roomId, setRoomId] = useState<string>('');
+  // const [mapImg, setMapImg] = useState<string>('');
   const [matchIndex, setMatchIndex] = useState<number[]>([]);
   const [isLoading, setIsLoading] = useState<boolean>(false);
   const [userName, setUserName] = useState<string>("");
+    const initialValues:tournament = {
+    gameName:"", mapType:"", gameType:"", version:"", roomUuid:"",  dateAndTime:new Date(), lastsurvival:"" , roomId:"", mapImg:""
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: lastsurvival should be came case
lastSurvival

File: src/app/userDashboard/tournament/page.tsx [hunk 23]
```diff
@@ -111,22 +99,27 @@ function Tournament() {
 
   useEffect(() => {
     setLastTournament(alldata[alldata.length - 1]);
-    setAllTournaments(alldata?.slice(0, 8));
+    setAllTournaments(alldata?.slice(0, 50));
     getRegisteredMatches();
-    getRoomidPwd();
+   // getRoomidPwd();
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: remove comments

File: src/app/userDashboard/tournament/page.tsx [hunk 24]
```diff
@@ -111,22 +99,27 @@ function Tournament() {
...
-      setMapImg(lastTournament?.mapImg);
+      setMatchDetails({
+        ...matchDetails,
+        gameName: lastTournament?.gameName,
+        gameType: lastTournament?.gameType,
+        mapType: lastTournament?.mapType,
+        version: lastTournament?.version,
+        roomId: lastTournament?.roomId,
+        dateAndTime: format,
+        lastsurvival: lastTournament?.lastsurvival,
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: lastsurvival is lastSurvival

File: src/app/userDashboard/tournament/page.tsx [hunk 25]
```diff
@@ -135,23 +128,20 @@ function Tournament() {
     gType: string,
     mType: string,
     vType: string,
-    mdate: string,
-    mtime: string,
-    lastServival: string,
+    dateandtime: Date,
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: dateandtime is dateAndTime camel casing

File: src/app/userDashboard/tournament/page.tsx [hunk 26]
```diff
@@ -135,23 +128,20 @@ function Tournament() {
...
+    roomUid: string,
+    lastsurvival: string,
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: lassurvival is lastSurvival

File: src/app/userDashboard/tournament/page.tsx [hunk 27]
```diff
@@ -442,11 +433,12 @@ function Tournament() {
...
                                 style={{ color: 'rgba(255, 122, 0, 1)' }}
                               >
-                                {mapType}
+                                {matchDetails?.mapType}
                               </span>
                             </div>
                           </div>
                           <div className={styles.spot_line_sec}>
+
                             {/* <progress
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx:422]: Either remove or uncomment

File: src/app/userDashboard/tournament/page.tsx [hunk 28]
```diff
@@ -489,15 +481,14 @@ function Tournament() {
                                       e.gameType,
                                       e.mapType,
                                       e.version,
-                                      e.date,
-                                      e.time,
+                                      new Date(e.dateAndTime), 
                                       e.lastServival,
```
Comment (expertdeveloper5821) [inline src/app/userDashboard/tournament/page.tsx]: check survival spelling

File: src/app/userDashboard/tournament/page.tsx (hunk 27 above)
Comment (deeksha-developer) [reply src/app/userDashboard/tournament/page.tsx:422]: i need this comment code

**Pull request merged**